pygmt/helpers/tempfile.py

```diff
@@ -17,12 +18,10 @@ def unique_name():
     Returns
     -------
     name : str
-        A unique name generated by ``tempfile.NamedTemporaryFile``
+        A unique name generated by ``uuid.uuid4``
 
     """
-    # Use the tempfile module to generate a unique name.
-    with NamedTemporaryFile(prefix="gmt-python-") as tmpfile:
-        return os.path.split(tmpfile.name)[-1]
+    return "gmt-python-" + uuid.uuid4().hex
```

Comment:
I'm not sure how to deal with the prefix `gmt-python`. Possible ways:
1. No prefix at all
2. Change `gmt-python-` to `pygmt-`
3. Add arguments `prefix` and `suffix` to `unique_name`.

Reply:
I think we can remove that prefix. Not sure why I added it but it really doesn't make a difference.